File: Dataset/Leetcode/test/62/456.cpp
class Solution {
public:
//计算n个里面选m个的组合数
int  countCombination(int n,int m){
    if(m==0||m==n) return 1;
    vector<vector<int> > dp(n+1,vector<int>(m+1,0));
    for(int i=0;i<=n;++i)
        dp[i][0]=1;  //m==0时为1
        
    for(int i=1;i<=n;++i){
        for(int j=1;j<=m;++j){
            if(i==j) dp[i][j]=1;
            else
                dp[i][j]=dp[i-1][j]+dp[i-1][j-1];
        }
    }
    return dp[n][m];
}
    int XXX(int m, int n) {
        return countCombination(m+n-2,m-1);  
    }
};

File: Dataset/Leetcode/train/125/456.py
 class Solution:
    def XXX(self, s: str) -> bool:
        arr =[i.lower() for i in s if i.isalpha() or i.isdigit()]
        if len(arr)%2==0:
            left = (len(arr)-1)//2
            right = (len(arr))//2
        else:
            left = right = (len(arr))//2
        while left>-1 and right<len(arr) and arr[left] ==arr[right]:
            left-=1
            right+=1
        if left==-1 and right==len(arr):return True
        else:return False

